File: company-service/src/main/java/ru/panic/companyservice/template/service/CompanyService.java
package ru.panic.companyservice.template.service;

import ru.panic.companyservice.template.dto.CreateCompanyRequest;
import ru.panic.companyservice.template.dto.CreateCompanyResponse;
import ru.panic.companyservice.template.entity.Company;

import java.util.List;

public interface CompanyService {
    List<Company> readAllCompanyByOwner(String jwtToken);
    CreateCompanyResponse createCompany(String jwtToken, CreateCompanyRequest companyRequest);
    void deleteCompany(String jwtToken, Company company);
}

File: company-service/src/main/kotlin/ru/panic/companyservice/template/service/CompanyService.kt
package ru.panic.companyservice.template.service

import ru.panic.companyservice.template.dto.CreateCompanyRequest
import ru.panic.companyservice.template.dto.CreateCompanyResponse
import ru.panic.companyservice.template.dto.DeleteCompanyRequest
import ru.panic.companyservice.template.entity.Company

interface CompanyService {
    fun createCompany(jwtToken: String, createCompanyRequest: CreateCompanyRequest): CreateCompanyResponse
    fun readAllByUsername(jwtToken: String): List<Company>?
    fun deleteByCompanyName(jwtToken: String, deleteCompanyRequest: DeleteCompanyRequest)
}